### PR TITLE
add test case for issue 106277 (PR 106268)

### PR DESCRIPTION
**Pull request by philipatl**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Adds test case for issue 106277 (PR 106268)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
> k8s.io/kubernetes/vendor/k8s.io/client-go/rest: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP2 expand_more	0s
> k8s.io/kubernetes/vendor/k8s.io/client-go/rest: TestHTTP1DoNotReuseRequestAfterTimeout expand_more

:thinking: :eyes: 
/retest

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go [hunk 1]
```diff
@@ -277,3 +281,241 @@ func TestManagedFieldsExtractAndRestore(t *testing.T) {
...
+	},
+}
+
+func TestEditOptions_visitToPatch(t *testing.T) {
+
+	expectedErr := func(err error) bool {
+		return err != nil && strings.Contains(err.Error(), "At least one of apiVersion, kind and name was changed")
+	}
+
+	type fields struct {
```
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go:316]: I think this struct can be dropped entirely

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go [hunk 2]
```diff
@@ -277,3 +281,241 @@ func TestManagedFieldsExtractAndRestore(t *testing.T) {
...
+					},
+				},
+				results: &editResults{},
+			},
+			checkErr: expectedErr,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := &EditOptions{
```
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go:495]:
```suggestion
			o := &EditOptions{}
```

**Review by eddiezane — COMMENTED** [review]
Thanks for adding these! I think we can do a bit of cleanup of unused overrides?

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go (hunk 2 above)
Comment (philipatl) [reply staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go]: @eddiezane Thanks, I've made the suggested changes.

**Review by eddiezane — APPROVED** [review]
Thanks!

/lgtm
/approve
/label tide/merge-method-squash

**Pull request merged**